src/main/java/pl/nogacz/snake/board/Board.java

```diff
@@ -116,6 +174,98 @@ private void moveSnakeHead(Coordinates coordinates) {
         }
     }
 
+    private void spawnRottenApple(){
+        if(Math.abs(lastSpawnTime - System.currentTimeMillis())/1000 == randSpawnRottenAppleTime){
+            addRottenApples();
+            rottenApples.add(currentRottenAppleCoordinates[0]);
+            lastSpawnTime = System.currentTimeMillis();
+            randSpawnRottenAppleTime = random.nextInt(9)+1;
+            //System.out.println("random spawn time:" + randSpawnRottenAppleTime);
+        }
+
+        if(Math.abs(lastSpawnTime2 - System.currentTimeMillis())/1000 == randSpawnRottenAppleTime2){
+            addRottenApples2();
+            rottenApples2.add(currentRottenAppleCoordinates[1]);
+            lastSpawnTime2 = System.currentTimeMillis();
+            randSpawnRottenAppleTime2 = random.nextInt(20)+1;
+            //System.out.println("random spawn time2:" + randSpawnRottenAppleTime2);
+        }
+
+        if(Math.abs(lastSpawnTime3 - System.currentTimeMillis())/1000 == randSpawnRottenAppleTime3){
+            addRottenApples3();
+            rottenApples3.add(currentRottenAppleCoordinates[2]);
+            lastSpawnTime3 = System.currentTimeMillis();
+            randSpawnRottenAppleTime3 = random.nextInt(15)+1;
+           //System.out.println("random spawn time3:" + randSpawnRottenAppleTime3);
+        }
+    }
+
+    private void disappearRottenApple(){
+        if(rottenApples.size() == 1 && disappearRotten){ //for initial dissappear process apple1.
+            randDissappearRottenAppleTime = random.nextInt(9) + 1;
+            //System.out.println("random dissappear time:" + randDissappearRottenAppleTime);
+            lastDissappearTime = System.currentTimeMillis();
+            disappearRotten=false;
+
+        }
+
+        if(rottenApples2.size() == 1 && disappearRotten2){ //for initial dissappear process apple2.
+            randDissappearRottenAppleTime2 = random.nextInt(9) + 1;
+            //System.out.println("random dissappear time2:" + randDissappearRottenAppleTime2);
+            lastDissappearTime2 = System.currentTimeMillis();
+            disappearRotten2=false;
+        }
+
+        if(rottenApples3.size() == 1 && disappearRotten3){ //for initial dissappear process apple3.
+            randDissappearRottenAppleTime3 = random.nextInt(10) + 1;
+            //System.out.println("random dissappear time3:" + randDissappearRottenAppleTime3);
+            lastDissappearTime3 = System.currentTimeMillis();
+            disappearRotten3=false;
+        }
+
+
+        if(Math.abs(lastDissappearTime - System.currentTimeMillis())/1000 == randDissappearRottenAppleTime){//apple1

```

Comment:
Bu dosyadaki birçok if conditonı çok uzun ve kodu anlamayı zorlaştırıyor. Bunları elden geçir ve iyileştir. Reuse edebileceğin bir durum varsa metot yazıp heryeden aynı metodu çağırarak reuse yap. Ya da her biri için manalı isimli bir metot yaz ve onu çağır.

Reply:
done